esmvalcore/local.py

```diff
@@ -422,110 +397,82 @@ def _select_drs(input_type, drs, project):
+def _get_globs(variable):
+    """Compose the globs that will be used to look for files."""
     project = variable['project']
 
-    root = get_rootpath(rootpath, project)
-    path_template = _select_drs('input_dir', drs, project)
+    rootpaths = _get_rootpath(project)
 
-    dirnames = []
-    for dirname_template in _replace_tags(path_template, variable):
-        for base_path in root:
-            dirname = os.path.join(base_path, dirname_template)
-            dirname = _resolve_latestversion(dirname)
-            if dirname is None:
-                continue
-            matches = glob.glob(dirname)
-            matches = [match for match in matches if os.path.isdir(match)]
-            if matches:
-                for match in matches:
-                    dirnames.append(match)
-            else:
-                logger.debug("Skipping non-existent %s", dirname)
+    dirname_template = _select_drs('input_dir', project)
+    dirname_globs = _replace_tags(dirname_template, variable)
 
-    return dirnames
+    filename_template = _select_drs('input_file', project)
+    filename_globs = _replace_tags(filename_template, variable)
 
+    globs = sorted(r / d / f for r in rootpaths for d in dirname_globs
+                   for f in filename_globs)
+    return globs
 
-def _get_filenames_glob(variable, drs):
-    """Return patterns that can be used to look for input files."""
-    path_template = _select_drs('input_file', drs, variable['project'])
-    filenames_glob = _replace_tags(path_template, variable)
-    return filenames_glob
 
+def _get_input_filelist(variable):
+    """Return the full path to input files."""
+    variable = dict(variable)
+    if 'original_short_name' in variable:
+        variable['short_name'] = variable['original_short_name']
 
-def _find_input_files(variable, rootpath, drs):
-    """Find available input files.
+    globs = _get_globs(variable)
+    logger.debug("Looking for files matching %s", globs)
 
-    Return the files, the directory in which they are located in, and
-    the file name.
-    """
-    short_name = variable['short_name']
-    variable['short_name'] = variable['original_short_name']
-    input_dirs = _find_input_dirs(variable, rootpath, drs)
-    filenames_glob = _get_filenames_glob(variable, drs)
-    files = find_files(input_dirs, filenames_glob)
-    variable['short_name'] = short_name
-    return (files, input_dirs, filenames_glob)
+    files = list(Path(file) for glob_ in globs for file in glob(str(glob_)))
+    files.sort()  # sorting makes it easier to see what was found
 
+    if 'timerange' in variable:
+        files = _select_files(files, variable['timerange'])
 
-def get_input_filelist(variable, rootpath, drs):
-    """Return the full path to input files."""
-    # change ensemble to fixed r0i0p0 for fx variables
-    # this is needed and is not a duplicate effort
-    if variable['project'] == 'CMIP5' and variable['frequency'] == 'fx':
-        variable['ensemble'] = 'r0i0p0'
```

Comment:
I started this review before your latest commits, sorry. Still one question: is this change reflected in the tests?

Reply:
I moved it to https://github.com/ESMValGroup/ESMValCore/blob/53f22a9a0956ea7f965c7e0272c0003675675c6e/esmvalcore/_recipe.py#L573-L574 because I think it is cleaner to modify the facets that are loaded from the recipe as appropriate when parsing the recipe instead of silently doing so when looking for files.

The tests in data_finder.yml have been updated accordingly.